nicegui/testing/user.py

```diff
@@ -33,14 +34,16 @@ def __init__(self, client: httpx.AsyncClient) -> None:
         self.forward_history: List[str] = []
         self.navigate = UserNavigate(self)
         self.notify = UserNotify()
+        self.download = UserDownload(self)
 
     def __getattribute__(self, name: str) -> Any:
-        if name not in {'notify', 'navigate'}:  # NOTE: avoid infinite recursion
+        if name not in {'notify', 'navigate', 'download'}:  # NOTE: avoid infinite recursion
             ui.navigate = self.navigate
             ui.notify = self.notify
+            ui.download = self.download
```

Comment:
How does this work? Does it monkeypatch the `ui` module?

If so, I'm wondering - what happens if the user does e.g. `from ui import download`? Wouldn't that break testing with `User`?

Reply:
You are right. The productive code must use `ui.download` for this to work. This fails:

```py
from nicegui import ui
from nicegui.testing import User
from nicegui.ui import download

pytest_plugins = ['nicegui.testing.user_plugin']


async def test_download(user: User):
    ui.button('get', on_click=lambda: download(b'some data'))

    await user.open('/')
    user.find('get').click()
    response = await user.download.next()
    assert response.text == 'some data'
```

I think we should not solve this in this PR. Other features like `ui.notify` or `ui.navigate` have the same problem. Maybe you have an idea on how we could improve this? Could you create a feature request?